drafts/taking-a-byte-of-the-carbon-cake.md

```diff
@@ -0,0 +1,291 @@
+Memory carbon rate = 0.0001 * 15 * 3 nodes = 0.0045 per hour
+Storage carbon rate, HDD = 0.0000677 * (100/1000) = 0.0000609 per hour
+Storage carbon rate, SSD = 0.0003049 * (100/1000) = 0.0027441 per hour
+Total rate for microservices = 0.01488 per hour
+
+Video platform:
+CPU carbon rate = 0.0006316 * 2 CPU * 2 nodes)  = 0.0025264 per hour
+Memory carbon rate = 0.0001 * 7.5 * 2 nodes = 0.0015 per hour
+Storage carbon rate, HDD = 0.0000677 * ( 20/1000 ) = 0.000001354 per hour
+Storage carbon rate, SSD =  0.000304 * ( 100/1000 ) = 0.0003049 per hour
+Total rate for video =  0.004058 per hour
+
+Database:
+CPU carbon rate = 0.0006316 * 1 node)  = 0.0006316 per hour
+Memory carbon rate = 0.0001 * 3.75  = 0.000375 per hour
+Storage carbon rate, SSD = ( 10/1000 * 0.0003049) = 0.00003049 per hour
+Total rate for database=  0.001037 per hour
+
+Total system carbon footprint for single environment =  (0.01488 + 0.004058 + 0.001037) * 8760 hours/year =~ 175 KgCO2/year
+Total system carbon footprint for 3 environments = 175 * 3 = 525 KgCO2/year
+
+```
+![Figure: Relative Carbon Footprints, showing our Example Micro-service Cloud System](https://cdn.hashnode.com/res/hashnode/image/upload/v1730817205560/Y1ZfUGOdl.jpg?auto=format)
+*Figure: Relative Carbon Footprints, including our Example Micro-service Cloud System*
+
+Let’s place our micro-services cloud system on the relative carbon footprint line. At 175 KgCO2 it would be somewhere in the middle between running a lawn mower and owning a cat. 
+But of course any realistic software system will likely have 2-3 similar environments, with varied configurations, plus services such as gateways, load balances, databases etc...
+These services may be PAYG shared, rather than dedicated to this system, but nevertheless need to be accounted. 
+
+In practice, it is likely that this carbon cost will be a multiple, let’s say 3x175 = 522 KgCO2/year for our example. This system is small scale in comparison to many brand-name e-commerce retail sites or banking systems, yet already close to the cost of a flight from London to Malaga at 620 KgCO2.
+
+
+## Green washing?
+
+Any article on carbon footprint is almost obliged to mention 'green washing', the subtle art of hiding the truth whilst appearing transparent on climate change and carbon footprints.
+
+Amazon has a target of 100% green by their own measurement. 
+That does NOT mean that your system will be carbon-friendly on AWS, it simply means that AWS are accounting for the offsets. 
+In regions where they are truly powered by green energy, that same power could be used elsewhere other than their data centre, and what incentive is there to decrease their (or our) usage?   
+
+Ultimately, the green software foundation's [GSF energy efficiency policy](https://learn.greensoftware.foundation/energy-efficiency) suggests we should aim to use less energy, and that has to be the end goal. Green data centres do not equal carbon-free environments in which to run our software.
```

Comment:
this IS an important information.

maybe instead of green washing, dive into the question whether all would be fine if we just use green energy...